src/index.ts

```diff
@@ -1,81 +1,155 @@
 import * as WebSocket from 'ws';
 import KaiEventEmitter from './KaiEventEmitter';
+import * as sdk from './sdk'
+import { KaiCapabilities } from "./KaiCapabilities";
+import { type } from 'os';
 
-let ws: WebSocket;
+let ws = new WebSocket('ws://localhost:2203');
+export let kaiEvents = new KaiEventEmitter();
+export let kaiCapabilities = KaiCapabilities;
 
-ws.on('open', () => {
-	// TODO:
-	// Test compartibility
-	// Send capabilities
+
+ws.on('message',function(data){
+    let response:sdk.Response = JSON.parse(JSON.parse(JSON.stringify(data)));
+    //console.log(response);
+    if(response['success']){
+        switch(response['type']){
+            case 'authentication':
+                kaiEvents.emit('authentication',response.success)      
+                break;
+            case 'setCapabilities':
+                kaiEvents.emit('setCapabilities','Capabilities Set')
+                break;
+            case 'getCapabilities':
+                kaiEvents.emit('getCapabilities',response)
+                break;
+            case 'fingerCalibration':
+                kaiEvents.emit('fingerCalibration',response)
+                break;  
+            case 'imuCalibration':
+                kaiEvents.emit('imuCalibration',response)
+                break;
+            case 'listConnectedKais':
+                kaiEvents.emit('listConnectedKais',response)
+                break;    
+            case 'getKaiData':
+                kaiEvents.emit('getKaiData',response)
+                break; 
+            case 'switchHand':
+                kaiEvents.emit('switchHand',response)
+                break;  
+            case 'incomingData':
+                kaiEvents.emit('incomingData',response) 
+                break;       
+            default:
+                kaiEvents.emit('default','Invalid response')
+                break;
+        };
+    }else{
+        kaiEvents.emit(response.type,response.error)
+    }
 });
 
-ws.on('message', (data) => {
-	var input = JSON.parse(data.toString());
-	KaiEvents.emit('data', input);
+//TODO:-
+export function setCapabilities(kaiId:number|"default"|"defaultLeft"|"defaultRight",capabilitiesArr:string[]){ 
```

Comment:
Why have you defined the **type** of the **kaiId** parameter over here, import all types from the **model** (here, _sdk.ts_). Import the type from _sdk.ts_ and use it.